core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -158,9 +211,17 @@ private int termLen(String text) {
     }
 
     public String asString() {
+        if (isAutoWithCustomValues()) {
+            return fuzziness.toString() + ":" + lowDistance + "," + highDistance;
+        }
         return fuzziness.toString();
     }
 
+    private boolean isAutoWithCustomValues() {
+        return fuzziness.startsWith("AUTO") && lowDistance != DEFAULT_LOW_DISTANCE &&
+            highDistance != DEFAULT_HIGH_DISTANCE;
```

Comment:
this will return false if the low distance is the same as the default low distance but the high distance is different from the default high distance, which sounds buggy?